PlutoHelperAgent/AppDelegate.m

```diff
@@ -193,7 +221,52 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
         BOOL extensionGood = [WhiteListProcessor checkIsInWhitelist:projectPath whiteListName:@"extensions_list" prefix:false];
               
         if (pathGood && extensionGood) {
-            [self tryOpenProject:projectPath];
+            NSArray *pathParts = [projectPath componentsSeparatedByString:@"?"];
+            NSString *partZero = [pathParts objectAtIndex:0];
+            if ((urlParams[@"premiereVersion"]) && (!urlParams[@"force"])) {
```

Comment:
I would start by putting this _whole_ if branch into its own function - taking in the `premiereVersion` and `force` params as arguments and then break it up from there; possibly in its own object

Reply:
You later requested something different, so I have not done this.